community/kernel-api/src/main/java/org/neo4j/internal/kernel/api/Kernel.java

```diff
@@ -29,4 +29,6 @@
     CursorFactory cursors();
 
     Session beginSession( SecurityContext securityContext );
+
+    Modes modes();
```

Comment:
What are we anticipating that we are going to do based on this `Modes`?
Some documentation about its _impact_ might be good.

Reply:
There's some documentation on the actual Modes method, not enough? Can write some more on the next drive by.